modules/internal/BytesStringCommon.chpl

```diff
@@ -719,27 +722,95 @@ module BytesStringCommon {
         const localNeedle = needle.localize();
         const needleLen = localNeedle.buffLen;
 
-        // i *is not* an index into anything, it is the order of the element
-        // of view we are searching from.
-        const numPossible = xLen - nLen + 1;
-        const searchSpace = if fromLeft
-            then 0..#(numPossible)
-            else 0..#(numPossible) by -1;
-        for i in searchSpace {
-          // j *is* the index into the localNeedle's buffer
-          for j in 0..#nLen {
-            const idx = view.orderToIndex(i+j); // 0s based idx
-            if x.buff[idx] != localNeedle.buff[j] then break;
-
-            if j == nLen-1 {
-              if count {
-                localRet += 1;
-              } else { // find
-                localRet = view.orderToIndex(i);
+        const numChar = 256;
+
+        if nLen <= 10 || xLen <= 100 {
```

Comment:
We should probably make these values `config const`s. I would imagine them being system-dependent, and it would be bad to see a user suffering from these hard-coded threshold and not being able to tweak them.

Ideally those should be documented, too, but for now let's not do that and keep them hidden.

I recommend names `chpl_bmhNeedleThreshold` and `chpl_bmhStringThreshold`, where `chpl_` prefix tells that these are not necessarily user-facing[, yet].